guardrails/validator_base.py

```diff
@@ -452,6 +465,38 @@ def validate(self, value: Any, metadata: Dict[str, Any]) -> ValidationResult:
         """Validates a value and return a validation result."""
         raise NotImplementedError
 
+    def validate_stream(self, chunk:Any, metadata: Dict[str, Any]) -> ValidationResult:
+        """Validates a chunk emitted by an LLM.
+        If the LLM chunk is smaller than the validator's chunking strategy, 
+        it will be accumulated until it reaches the desired size. In the meantime, 
+        the validator will return None.
+
+        Otherwise, the validator will validate the chunk and return the result.
+        """
```

Comment:
Definitely not a right now thing, but we should look at ways to only yield when there's something _to_ yield.  My immediate thought it to use an event system when the Runner is the listener; this way we would only emit and event with usable information instead of yielding `None`.  Another way that might also work is the same concept but with callbacks; i.e. don't call the callback unless there's something to call it with.  TL;DR; if we can break out of an async/await type of structure where you have to yield/return something every iteration it would produce a more intuitive output.